### PR TITLE
feat: Select Engine versions

### PR DESCRIPTION
**Pull request by arcoraven**
https://linear.app/thirdweb/issue/INFRA-401/update-multiple-engine-versions

image

image

---

## PR-Codex overview
This PR primarily refactors the handling of `teamSlug` across various components and introduces new functionalities related to deployment public configuration and version updates.

### Detailed summary
- Renamed `team_slug` to `teamSlug` in multiple components.
- Added `useEngineGetDeploymentPublicConfiguration` hook for fetching deployment config.
- Updated `EngineVersionBadge` and `ChangeVersionModal` to accommodate new version update logic and alerts.
- Modified the update process to include team-specific deployment paths.
- Enhanced user interface with alerts and new selection options for version updates.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: apps/dashboard/src/app/team/[team_slug]/(team)/~/engine/(instance)/[engineId]/_components/EnginePageLayout.tsx [hunk 1]
```diff
@@ -261,7 +261,7 @@ function RenderEngineInstanceHeader(props: {
             )}
           </div>
         </div>
-        <EngineVersionBadge instance={instance} />
+        <EngineVersionBadge instance={instance} teamId={instance.accountId} />
```
Comment (MananTank) [inline apps/dashboard/src/app/team/[team_slug]/(team)/~/engine/(instance)/[engineId]/_components/EnginePageLayout.tsx:264]: Please use `team.id` here

**Review by MananTank — CHANGES_REQUESTED** [review]
please address comments

File: apps/dashboard/src/app/team/[team_slug]/(team)/~/engine/(instance)/[engineId]/_components/version.tsx [hunk 2]
```diff
@@ -18,37 +10,73 @@ import {
...
+    teamId,
+  });
   const [isModalOpen, setModalOpen] = useState(false);
 
-  const currentVersion = healthQuery.data?.engineVersion ?? "...";
-  const latestVersion = latestVersionQuery.data;
-  const isStale = latestVersion && currentVersion !== latestVersion;
+  if (
+    !healthQuery.data ||
+    publicConfigurationQuery.data.serverVersions?.length === 0
```
Comment (MananTank) [inline apps/dashboard/src/app/team/[team_slug]/(team)/~/engine/(instance)/[engineId]/_components/version.tsx:54]: Build is failing. Type issue: 
https://vercel.com/thirdweb/thirdweb-www/C2i1x6cZEVJuKfQSe6uz45NXdhnv?filter=errors

**Review by MananTank — APPROVED** [review]

**Comment by arcoraven** [comment]
### Merge activity

* **Nov 25, 7:53 AM EST**: The merge label 'merge-queue' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=thirdweb-dev&repo=js) once it meets the requirements.
* **Nov 25, 7:53 AM EST**: A user added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=thirdweb-dev&repo=js).

**Pull request merged**